tests/trace_server_bindings/test_async_batch_processor.py

```diff
@@ -60,3 +60,195 @@ def test_wait_until_all_processed():
     # We should only see the first batch.  Everything else is stuck in the queue.
     processor_fn.assert_has_calls([call([1, 2, 3])])
     assert processor.queue.qsize() == 6
+
+
+def test_health_check_thread_is_created():
+    """Test that health check thread is created and running."""
+    processor_fn = MagicMock()
+    processor = AsyncBatchProcessor(
+        processor_fn, max_batch_size=100, min_batch_interval=0.1
+    )
+
+    # Health check thread should be alive
+    assert processor.health_check_thread.is_alive()
+    assert processor.processing_thread.is_alive()
+
+    processor.stop_accepting_new_work_and_flush_queue()
+
+    assert not processor.health_check_thread.is_alive()
+    assert not processor.processing_thread.is_alive()
+
+    # restart thread
+    processor.accept_new_work()
+    assert processor.health_check_thread.is_alive()
+    assert processor.processing_thread.is_alive()
+    processor.stop_accepting_new_work_and_flush_queue()
+
+
+def test_health_check_functionality_and_logging():
+    """Test health check revival functionality, logging, and full queue behavior."""
+    processor_fn = MagicMock()
+
+    with patch(
+        "weave.trace_server_bindings.async_batch_processor.logger"
+    ) as mock_logger:
+        # Create processor with small queue to test full queue behavior
+        processor = AsyncBatchProcessor(
+            processor_fn, max_batch_size=100, min_batch_interval=0.1, max_queue_size=1
+        )
+
+        # Test 1: Test the _ensure_health_check_alive method directly
+        original_health_thread = processor.health_check_thread
+        with patch.object(
+            processor.health_check_thread, "is_alive", return_value=False
+        ):
+            processor._ensure_health_check_alive()
+
+            # Should have created a new health check thread
+            assert processor.health_check_thread != original_health_thread
+            assert processor.health_check_thread.is_alive()
+            mock_logger.warning.assert_called_with(
+                "Health check thread died, attempting to revive it"
+            )
+            mock_logger.info.assert_called_with(
+                "Health check thread successfully revived"
+            )
+
+        # Test 2: Health check revival on full queue
+        mock_logger.reset_mock()
+        with patch.object(
+            processor.health_check_thread, "is_alive", return_value=False
+        ):
+            processor.enqueue([1])  # Fills the queue
+            processor.enqueue(
+                [2]
+            )  # Should trigger Full exception and health check revival
```

Comment:
Then why is there no `with pytest.raises` here?

Reply:
Dumb comment, we catch on `Full` and log the warning, which gets asserted on in other tests in `test_weave_client.py`